### DISCUSSION TITLE
RFE: libcamera scaling instead of cropping

### DISCUSSION THREAD

**Discussion by wolfgangrupprecht** [Motion-Project/motionplus#168] [Enhancements]
Under the old RPI camera interface setting a width and height in motion(1) used to scale the full camera frame down to the requested size.  Not only did this shrink the stored image's file size but it also improved the quality of the picture by averaging the noise across all the sensor cells that were combined into the shrunken image's pixels.   Under libcamera this is no longer the case.   The full frame is simply cropped to generate a cropped image from the center of the frame.   The resultant image has both tunnel-vision, missing most of the frame and also as a bonus has downright awful pixelation and sensor noise.   I realize this is rightfully a libcamera issue.  They took the easy way out to reduce the image size but the bottom line is that the images from native RPI cameras now suck.  The default 640x480 image only records an area of 1/25th of the full frame for v2 camera modules, if I did the math right.  The image quality is awful.   Is there a work-around that folks are using?

**Comment by sroller** [comment]
Do you know if this has been addressed?
